src/constructs/autoscaling/asg.ts

```diff
@@ -21,6 +33,53 @@ export interface GuAutoScalingGroupProps
   overrideId?: boolean;
 }
 
+type GuStageDependentAsgProps = Record<Stage, GuAsgCapacityProps>;
+
+/**
+ * `minimumInstances` determines the number of ec2 instances running under normal circumstances
+ * (i.e. when there are no deployment or scaling events in progress).
+ *
+ * `maximumInstances` is optional. If omitted, this will be set to `minimumInstances * 2`.
+ * This allows us to support Riff-Raff's autoscaling deployment type by default.
+ *
+ * The maximum capacity value should only be set if you need to scale beyond the default limit (e.g. due to heavy traffic)
+ * or restrict scaling for a specific reason.
+ */
+export interface GuAsgCapacityProps {
+  minimumInstances: number;
+  maximumInstances?: number;
+}
+
+interface AwsAsgCapacityProps {
+  minCapacity: number;
+  maxCapacity: number;
+}
+
+function wireStageDependentProps(stack: GuStack, stageDependentProps: GuStageDependentAsgProps): AwsAsgCapacityProps {
+  const minInstancesKey = "minInstances";
+  const maxInstancesKey = "maxInstances";
+  const minInstances: GuStageDependentValue<number> = {
+    variableName: minInstancesKey,
+    stageValues: {
+      [Stage.CODE]: stageDependentProps.CODE.minimumInstances,
+      [Stage.PROD]: stageDependentProps.PROD.minimumInstances,
+    },
+  };
+  const maxInstances: GuStageDependentValue<number> = {
+    variableName: maxInstancesKey,
+    stageValues: {
+      [Stage.CODE]: stageDependentProps.CODE.maximumInstances ?? stageDependentProps.CODE.minimumInstances * 2,
```

Comment:
minor: would be good to do some validation here to assert min > max and throw otherwise. Maybe for a new PR.

Reply:
I will add this in separate PR (I don't want to hold up the queue of PRs which edit this file!)